openhcl/hcl/src/ioctl.rs

```diff
@@ -1691,178 +1891,144 @@ impl<'a, T: Backing> ProcessorRunner<'a, T> {
         &mut self,
-        name: HvX64RegisterName,
-        value: HvRegisterValue,
+        #[cfg(guest_arch = "x86_64")] names: &[HvX64RegisterName],
+        #[cfg(guest_arch = "aarch64")] names: &[HvArm64RegisterName],
+        values: &mut [HvRegisterValue],
+        vtl: GuestVtl,
     ) -> Result<(), Error> {
-        tracing::trace!(?name, ?value, "set_vp_register");
-        self.set_vp_register_inner(name.into(), value)
+        self.get_vp_registers_inner(names, values, vtl)
     }
 
-    /// Set the following register on the given VP, aarch64
+    /// Set the following register on the current VP.
     ///
     /// This will fail for registers that are in the mmapped CPU context, i.e.
     /// registers that are shared between VTL0 and VTL2.
-    ///
-    /// Can only set registers for VTL 0.
-    #[cfg(guest_arch = "aarch64")]
     pub fn set_vp_register(
         &mut self,
-        name: HvArm64RegisterName,
+        #[cfg(guest_arch = "x86_64")] name: HvX64RegisterName,
+        #[cfg(guest_arch = "aarch64")] name: HvArm64RegisterName,
         value: HvRegisterValue,
+        vtl: GuestVtl,
     ) -> Result<(), Error> {
-        tracing::trace!(?name, ?value, "set_vp_register");
-        self.set_vp_register_inner(name.into(), value)
-    }
-
-    fn get_vp_register_inner(&mut self, name: HvRegisterName) -> Result<HvRegisterValue, Error> {
-        if let Some(value) = T::try_get_reg(self, name)? {
-            return Ok(value);
-        }
-        let mut info = [HvRegisterAssoc {
-            name,
-            pad: Default::default(),
-            value: FromZeroes::new_zeroed(),
-        }];
-        self.get_reg(&mut info)?;
-        Ok(info[0].value)
+        self.set_vp_registers([(name, value)], vtl)
     }
 
-    /// Get the following register on the current VP, x86_64.
-    ///
-    /// This will fail for registers that are in the mmapped CPU context, i.e.
-    /// registers that are shared between VTL0 and VTL2.
-    ///
-    /// Can only get registers for VTL 0.
-    #[cfg(guest_arch = "x86_64")]
-    pub fn get_vp_register(&mut self, name: HvX64RegisterName) -> Result<HvRegisterValue, Error> {
-        self.get_vp_register_inner(name.into())
-    }
-
-    /// Get the following register on the current VP, aarch64.
-    ///
-    /// This will fail for registers that are in the mmapped CPU context, i.e.
-    /// registers that are shared between VTL0 and VTL2.
-    ///
-    /// Can only get registers for VTL 0.
-    #[cfg(guest_arch = "aarch64")]
-    pub fn get_vp_register(&mut self, name: HvArm64RegisterName) -> Result<HvRegisterValue, Error> {
-        tracing::trace!(?name, "get_vp_register");
-        self.get_vp_register_inner(name.into())
-    }
-
-    /// Sets a set of VP registers on the last intercepting VTL.
-    pub fn set_vp_registers<I>(&mut self, values: I) -> Result<(), Error>
+    /// Sets a set of VP registers.
+    pub fn set_vp_registers<I>(&mut self, values: I, vtl: GuestVtl) -> Result<(), Error>
```

Comment:
On all of these, please move `vtl` to be just after `self`. This makes calling code much easier to read, because the list ends up at the end.

Reply:
Ditto everywhere, even if you're adding this to a function that doesn't take a list.